src/msgraph_core/tasks/large_file_upload.py

```diff
@@ -144,7 +119,15 @@ async def upload(self, after_chunk_upload: Optional[Callable] = None):
                 logging.error("Error uploading chunk  %s", error)
             finally:
                 self.chunks -= 1
-        return session
+        upload_result = UploadResult()
+        upload_result.upload_session = UploadSessionDataHolder(
+            expiration_date_time=self.upload_session.expiration_date_time,
+            next_expected_ranges=self.upload_session.next_expected_ranges,
+            upload_url=self.upload_session.upload_url
+        )
+        upload_result.item_response = session
+        upload_result.location = self.upload_session.upload_url
```

Comment:
The location should be updated with the value of the [Location header](https://learn.microsoft.com/en-us/graph/outlook-large-attachments?tabs=http#step-3-continue-uploading-byte-ranges-until-the-entire-file-has-been-uploaded) if it's present for Outlook uploads.